Article - Customize CSS using Bulma

cyberShaw opened the pull request:
Added article on the **Bulma CSS** framework.

This details on the need for CSS frameworks, the advantages of Bulma and how to customize your websites using Bulma. Please review and merge 😄

louisefindlay23 commented on articles/customization-using-bulma/index.md:
```diff
@@ -0,0 +1,233 @@
+published: true
+url: /engineering-education/customization-using-bulma/
+title: Customize CSS using Bulma
+description: Bulma is a lightweight, open source CSS framework that has excellent customization features, perfect for rapid development.
```
replace , with which is.

louisefindlay23 commented on articles/customization-using-bulma/index.md:
```diff
@@ -0,0 +1,233 @@
+
+**Cascading Style Sheets** is a stylesheet language that is used to describe how a webpage looks and feels visually. To speed up the development of projects, developers use CSS frameworks like Bulma. Let us see why CSS frameworks are useful and learn about Bulma.
+
+### Table of Contents
```
Headings should start at H2 onwards (##)

louisefindlay23 commented on articles/customization-using-bulma/index.md:
```diff
@@ -0,0 +1,233 @@
+- [Customizing Variables](#customizing-variables)
+- [Further Reading](#further-reading)
+
+### Need for a CSS Framework
```
Should add The before Need

louisefindlay23 commented on articles/customization-using-bulma/index.md:
```diff
@@ -0,0 +1,233 @@
+
+### Need for a CSS Framework
+
+CSS is an flexible language, with a ton of customization options. This may sometime prove to be overwhelming for a beginner. When you're out to build a product, a CSS framework helps to accelerate the process of development. It helps to reduce time to production and this may prove crucial to a project. Some other benefits to using a CSS framework include:
```
Consider revising to "This may sometime**s** be overwhelming" and revise time to production to aid flow.

louisefindlay23 commented on articles/customization-using-bulma/index.md:
```diff
@@ -0,0 +1,233 @@
+- It has an excellent developer community.
+- It can be customized extensively through variables.
+- Bulma uses easy-to-read class names that are easy to remember.
+- It is extremely clean and aesthetic looking.
```
It provides a clean and modern design would flow better.

louisefindlay23 commented on articles/customization-using-bulma/index.md:
````diff
@@ -0,0 +1,233 @@
+npm install bulma
+```
+
+- We can also download the `bulma.css` file from [here](https://github.com/jgthms/bulma/tree/master/css) and include it in our webpage using the `<link>` tag.
````
You should provide an example like the previous two options.

louisefindlay23 commented on articles/customization-using-bulma/index.md:
```diff
@@ -0,0 +1,233 @@
+
+### Components
+
+Bulma comes with many in-built UI components. Let us try to create a card component.
```
Let's would flow better here.

louisefindlay23 commented on articles/customization-using-bulma/index.md:
```diff
@@ -0,0 +1,233 @@
+- Generic Variables -- HTML elements that cary no CSS class.
+- Element Variables -- variables that are specific to a bulma element.
+
+Now, for customizing these Bulma Variables, let us set up sass and bulma.
```
Consider adding a brief explanation of Sass. Readers who are new to CSS frameworks most likely don't know what Sass is.

louisefindlay23 commented on articles/customization-using-bulma/index.md:
```diff
@@ -0,0 +1,233 @@
+
+- Initial Variables -- There are global variables with literal values.
+- Derived Variables -- These are derived from other variables.
+- Generic Variables -- HTML elements that cary no CSS class.
```
car**r**y

louisefindlay23 commented on articles/customization-using-bulma/index.md:
```diff
@@ -0,0 +1,233 @@
+  - style.css
+  - style.scss
+
+Let us now create a html page with a basic button.
```
HTML should be all caps.

louisefindlay23 commented on articles/customization-using-bulma/index.md:
````diff
@@ -0,0 +1,233 @@
+We can do this by modifying `style.scss`
+
+```scss
+// Importing Initital Variables
````
Initial

louisefindlay23 commented on articles/customization-using-bulma/index.md:
```diff
@@ -0,0 +1,233 @@
+$primary: $pink;
+$danger: #0000ec;
+
+// Import the rest of bulma
```
**B**ulma. Use consistent case (capitilisation)

louisefindlay23 reviewed (changes requested):
Great article, @cyberShaw. I've never used Bulma before but it sounds a great CSS framework. I've provided a few suggestions to aid readability and some corrections you should make.

hectorkambow commented:
Plagiarism check done

cyberShaw commented:
> Great article, @cyberShaw. I've never used Bulma before but it sounds a great CSS framework. I've provided a few suggestions to aid readability and some corrections you should make.

Thank you for the review and the valuable feedback, @louisefindlay23.
I've implemented the changes you've requested in e30d3e9😄

louisefindlay23 commented on articles/customization-using-bulma/index.md:
```diff
@@ -0,0 +1,237 @@
+- Generic Variables -- HTML elements that carry no CSS class.
+- Element Variables -- variables that are specific to a bulma element.
+
+Now, for customizing these Bulma Variables, let us set up sass and bulma. Sass is a CSS preprocessor that adds extra features like variables and loops. It is an extension to CSS that improves the stylesheet language and reduces repetition.
```
There should be consistent capitalisation on Sass and Bulma. You have various different cases of Bulma (lower-case and initial capitalised) throughout the article. 

I would recommend sticking to Bulma.

cyberShaw replied:
I've replaced all instances of Sass and Bulma with their appropriate capitalized forms in 5c68e73! 😄

louisefindlay23 commented on articles/customization-using-bulma/index.md:
````diff
@@ -0,0 +1,237 @@
+npm i bulma sass
+```
+
+- Now, to get started, create three files.
````
create the three files below would be more semantic.

cyberShaw replied:
Done, change made!

louisefindlay23 reviewed (changes requested):
Thanks for making the changes, @cyberShaw and I'm glad you find my feedback helpful. 

There's just one or two small grammatical errors you should fix before the article can be passed on to tech review.

louisefindlay23 reviewed (approved):
Thanks for making the changes, @cyberShaw. Please take these on board for future articles that you write. I've made some further minor changes and your article is now on to the tech review stage.

louisefindlay23 commented on articles/customization-using-bulma/index.md:
````diff
@@ -0,0 +1,234 @@
+
+- Download the npm package from the npm repository.
+
+```bash
````
You should mention that it is good practise to create a `package.json` file first using `npm init`.

cyberShaw commented:
> Thanks for making the changes, @cyberShaw. Please take these on board for future articles that you write. I've made some further minor changes and your article is now on to the tech review stage.

I'll keep these changes in mind and implement them next time. 😄

louisefindlay23 commented on articles/customization-using-bulma/index.md:
````diff
@@ -0,0 +1,234 @@
+
+```bash
+npm install bulma
+```
````
You need to add `import 'bulma/css/bulma.css'` in your CSS file otherwise Bulma will not work per the documentation.

louisefindlay23 replied:
Actually, it should be `@import "../node_modules/bulma/css/bulma.css";`. The previous line I gave you only works if you're using a bundler like Parcel.

cyberShaw replied:
Noted! :fire:

louisefindlay23 commented on articles/customization-using-bulma/index.md:
````diff
@@ -0,0 +1,234 @@
+npm install bulma
+```
+
+- We can also download the `bulma.css` file from [here](https://github.com/jgthms/bulma/tree/master/css) and include it in our webpage using the `<link>` tag.
````
That link is only to the CSS folder, not to the file that downloads it. I'd also recommend using the minified version. https://github.com/jgthms/bulma/blob/master/css/bulma.min.css

louisefindlay23 commented on articles/customization-using-bulma/index.md:
````diff
@@ -0,0 +1,234 @@
+For example:
+
+```html
+<div class="columns">
````
This did not work for me using the NPM version of Bulma.

cyberShaw replied:
> This did not work for me using the NPM version of Bulma.

Did you import the CSS? by running,
```css
import 'bulma/css/bulma.css'
```

louisefindlay23 replied:
I found out why it wasn't working, @cyberShaw. The line I gave you (which is in Bulma's documentation as how you should import the NPM module) only works if you're using a bundler. I've given you an updated line of code which works without one.

louisefindlay23 commented on articles/customization-using-bulma/index.md:
````diff
@@ -0,0 +1,234 @@
+This snippet would create a responsive column layout with three columns of equal width. You can also specify the column size by using the size classes.
+
+```html
+<div class="columns">
````
Consider adding screenshots for both column examples.

louisefindlay23 commented on articles/customization-using-bulma/index.md:
````diff
@@ -0,0 +1,234 @@
+$family-primary: "Times New Roman";
+```
+
+Now, the font of the document is changed to Times New Roman.
````
You should remind readers they have to run sass style.scss:style.css again.

louisefindlay23 reviewed (changes requested):
Just tech reviewed your article, @cyberShaw. Great use of screenshots but there were a few areas you should clarify more to readers and you're missing steps for installing Bulma using NPM.

cyberShaw commented:
Thank you for the tech review @louisefindlay23.
Implemented all requested changes 😄

louisefindlay23 reviewed (changes requested):
Thanks for making the changes, @cyberShaw. I've modified the code snippet I gave you so it should work on a vanilla NPM environment. When you can make that small change, I can approve your article for publication.

cyberShaw commented:
> Thanks for making the changes, @cyberShaw. I've modified the code snippet I gave you so it should work on a vanilla NPM environment. When you can make that small change, I can approve your article for publication.

Changed it as requested in 6fb78cf

louisefindlay23 reviewed (approved):
Thanks, @cyberShaw. This is ready for publication, @hectorkambow once I add peer reviewer credit.
